intersystems/git-source-control
# Suppress deletion in locked environment

Testing Steps:
1. In Terminal/Windows Powershell, clone git-source-control repository and switch to testDelete branch
```
git clone git@github.com:intersystems/git-source-control.git
```
(enter passcode for deploy key if prompted)
```
cd git-source-control
git checkout suppressDelete
```

In IRIS terminal
2. Load git-source-control repository into IRIS from the community package registry.
```
zpm "install git-source-control"
```
3. Configure settings by running the following method and answering the prompts:
```
zpm "q"
d ##class(SourceControl.Git.API).Configure()
```
4. Load git-source-control into the current namespace
```
zpm "load <path to local git-source-control repo cloned in step 1>"
```
5. Open Studio
If you see "Dubious Ownership" error in Output window, you have to change owner of local git-source-control repo. Steps to do that on Windows:
	1) Open Windows Powershell as Administrator
	2) Run 
	       a. `$objSID = New-Object System.Security.Principal.SecurityIdentifier("<SID>")`
                     Note: replace \<SID\> with \<SID\> in "the current user is: '\<SID\>'" part of dubious ownership error message
	       b. `$objUser = $objSID.Translate([System.Security.Principal.NTAccount])`
	       c. `icacls <path to local git-source-control repo cloned in step 1> /setowner $objUser.value /T `
		Note: replace <path to local git-source-control repo cloned in step 1> with a path like C:\Users\<your username>\ 
	3) Close and reopen Studio
6. Lock the environment
```
d ##class(SourceControl.Git.API).Lock()
```
7. In Studio's Workspace window Namespace tab, click Classes > SourceControl > Git > TestDelete.cls
8. Right-click on TestDelete and then click "Delete Class 'SourceControl.Git.TestDelete'" An error message should pop up with message "Can't delete in locked environment"
9. Open git-source-control repo in VS Code.

In VS Code,
10. Click the ObjectScript icon in the left side menu to open ObjectScript Explorer window, click "Choose Server and Namespace" and sign in using InterSystems Server Credentials.
11. Open an InterSystems Websocket Terminal.
12. Verify that you are still in locked environment
```
w ##class(SourceControl.Git.Utils).Locked()
```
should print 1
13. In ObjectScript Explorer window, click Classes > SourceControl > Git > TestDelete.cls
14. Right-click on TestDelete and then click "Delete" > "Confirm". Verify that TestDelete file wasn't deleted.
